### PR TITLE
refactor code in processors to add more readablity

### PR DESCRIPTION
**Pull request by allencloud**
I found it is a little bit hard to understand the related code.

So I try to refactor them to make them more readable.

Signed-off-by: allencloud <allen.sun@daocloud.io>

### CODE REVIEW

**Comment by allencloud** [comment]
any update? ☀️

File: metrics/processors/namespace_based_enricher.go [hunk 1]
```diff
@@ -46,28 +46,33 @@ func (this *NamespaceBasedEnricher) Process(batch *core.DataBatch) (*core.DataBa
 
 // Adds UID to all namespaced elements.
 func (this *NamespaceBasedEnricher) addNamespaceInfo(metricSet *core.MetricSet) {
-	if metricSetType, found := metricSet.Labels[core.LabelMetricSetType.Key]; found &&
-		(metricSetType == core.MetricSetTypePodContainer ||
-			metricSetType == core.MetricSetTypePod ||
-			metricSetType == core.MetricSetTypeNamespace) {
+	metricSetType, found := metricSet.Labels[core.LabelMetricSetType.Key]
+	if found {
```
Comment (DirectXMan12) [inline metrics/processors/namespace_based_enricher.go]: `if !found`

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
I'm generally in favor of this change (it brings the code more in line with general kubernetes coding conventions), but I found at least one logic error.  I'd like to get a second pair of eyes on this @piosz

**Comment by allencloud** [comment]
Thanks for your review. @DirectXMan12  I have updated the PR. Actually double review or more is necessary and first we should absolutely make the logic consistent.

**Review by DirectXMan12 — APPROVED** [review]
I believe this LGTM, but I'd like to get a couple other eyes on it @mwielgus @piosz @crassirostris

**Comment by allencloud** [comment]
Hi, @DirectXMan12 
Is it OK to attach a `ok-to-test` comment to this PR?
Since I think it will trigger another important test.  😄 ☀️

**Comment by DirectXMan12** [comment]
yep! :-)

/ok-to-test

**Comment by allencloud** [comment]
Could you help to review this PR? @mwielgus @piosz @crassirostris
Thanks a lot. 😃

**Comment by allencloud** [comment]
Could you help to review this PR? @mwielgus @piosz @crassirostris
And please feel free to tell me if I missed something.
Thanks a lot. 😃 🍻

**Comment by crassirostris** [comment]
Sorry, I don't have enough context and capacity right now

**Comment by piosz** [comment]
Thanks @allencloud for your contribution!

**Pull request merged**